### PR TITLE
Download handling and reporting fixes

### PR DESCRIPTION
**Pull request by rsalveti**

### CODE REVIEW

**Comment by gatek33p3r** [comment]
Checkpatch Suceeded.

**Comment by gatek33p3r** [comment]
Bootloader Build Succeeded.

**Comment by gatek33p3r** [comment]
Application Build Succeeded.

File: src/tcp.c [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 #define SERVER_CONNECT_TIMEOUT		K_SECONDS(5)
 #define SERVER_CONNECT_MAX_WAIT_COUNT	2
-#define TCP_TX_TIMEOUT			K_MSEC(500)
+#define TCP_TX_TIMEOUT			500 /* ms */
```
Comment (mike-scott) [inline src/tcp.c]: Why remove the K_MSEC wrapper?  It seemed like at one point Zephyr was leaning towards using it to clearly mark values as MS.  Tho currently it's just a stub:
#define K_MSEC(ms)     (ms)

**Comment by gatek33p3r** [comment]
Checkpatch Suceeded.

**Comment by gatek33p3r** [comment]
Bootloader Build Succeeded.

**Comment by gatek33p3r** [comment]
Application Build Succeeded.

**Pull request merged**